win32ss/user/user32/windows/cursoricon.c

```diff
@@ -1148,6 +1149,14 @@ BITMAP_LoadImageW(
         goto end;
     CopyMemory(pbmiCopy, pbmi, iBMISize);
 
+    /* Test compression BI_BITFIELDS & if its 12 bytes are in Resource Size */
+    if (pbmiCopy->bmiHeader.biSizeImage + pbmiCopy->bmiHeader.biSize + 12 == ResSize
```

Comment:
Is this not something that should be handled in `bitmap_info_size` ?

Reply:
Yes, I agree that it should. It's moved into this section of code now. I found that Wine's code now has a descriptive header for ```bitmap_info_size``` and I have included it now with revisions to show the accounting for the bit-fields count now. Also, now I have separated out the code to add these bytes and the code for correcting for GCC's improper handling of 32-bpp BI_BITFIELDS bitmaps.

Reply:
Here is the result of testing the new 16-bpp bitmap in MSVC with the new changes:

![wordpad_toolbar_16bpp_MSVC](https://github.com/reactos/reactos/assets/32620577/16470852-2831-4017-8841-9a274c4ab273)

Here is the result of testing the new 16-bpp bitmap in GCC with the new changes:

![wordpad_toolbar_16bpp_GCC](https://github.com/reactos/reactos/assets/32620577/deb4dab7-9357-4f27-9810-7bee1f3dddfd)